Azure/sonic-buildimage
# [security] Upgrade kernel to 3.16.57-2 on 201803 branch

**- What I did**
Upgrade kernel to 3.16.57-2 on 201803 branch
to fix security issues like:
    https://www.debian.org/security/2018/dsa-4120
    https://www.debian.org/security/2018/dsa-4179
    https://www.debian.org/security/2018/dsa-4187
    https://www.debian.org/security/2018/dsa-4188
    https://www.debian.org/security/2018/dsa-4196
    
and more.

**- How I did it**
Use the code in sonic-linux-kernel PR:
https://github.com/Azure/sonic-linux-kernel/pull/68

Fix all modules that rely on kernel modules at sonic-buildimage

**- How to verify it**
Built opennsl module locally with the sonic-linux-kernel code 3.16.57-2, point the mk file to that module. 

Build onie image and loaded to the box with BRCM chip:

sonic:~$uname -a
Linux lnos-x1-a-asw03 3.16.0-6-amd64 #1 SMP Debian 3.16.57-2 (2015-12-19) x86_64 GNU/Linux

sonic:~$ show interfaces status 
  Interface    Lanes    Speed    MTU    Alias    Oper    Admin
-----------  -------  -------  -----  -------  ------  -------
  Ethernet0       65      10G   9100   Eth1/1    down       up
  Ethernet1       66      10G   9100   Eth1/2    down       up
  Ethernet2       67      10G   9100   Eth1/3    down       up
  Ethernet3       68      10G   9100   Eth1/4    down       up
  Ethernet4       69      10G   9100   Eth2/1    down       up
  Ethernet5       70      10G   9100   Eth2/2    down       up
  Ethernet6       71      10G   9100   Eth2/3    down       up
  Ethernet7       72      10G   9100   Eth2/4    down       up
  Ethernet8       73      10G   9100   Eth3/1      up       up
  Ethernet9       74      10G   9100   Eth3/2      up       up
 Ethernet10       75      10G   9100   Eth3/3    down       up


Note: The opennsl module should be updated to MSFT official one before merge. Other ASIC vendors binary links need to be updated as well.


Test:

show ver
SONiC Software Version: SONiC.HEAD.3000-0a5337f
Distribution: Debian 8.11
Kernel: 3.16.0-6-amd64
Build commit: 0a5337f
Build date: Fri Oct  5 19:56:19 UTC 2018
Built by: johnar@jenkins-worker-1

----------------------------------------------------------------------------------------------------
Summary:
----------------------------------------------------------------------------------------------------
TestSuite: acl                  Pass: 276                  Fail: 0                   
TestSuite: arp                  Pass: 98                   Fail: 0                   
TestSuite: bgp_fact             Pass: 59                   Fail: 0                   
TestSuite: continuous_reboot    Pass: 112                  Fail: 0                   
TestSuite: everflow_testbed     Pass: 493                  Fail: 0                   
TestSuite: fib                  Pass: 67                   Fail: 0                   
TestSuite: lldp                 Pass: 62                   Fail: 0                   
TestSuite: link_flap            Pass: 564                  Fail: 0                   
TestSuite: mem_check            Pass: 56                   Fail: 0                   
TestSuite: mtu                  Pass: 62                   Fail: 0                   
TestSuite: port_toggle          Pass: 61                   Fail: 0                   
TestSuite: reboot               Pass: 69                   Fail: 0                   
TestSuite: repeat_harness       Pass: 61                   Fail: 0                   
TestSuite: restart_swss         Pass: 53                   Fail: 0                   
TestSuite: restart_swss_service Pass: 69                   Fail: 0                   
TestSuite: sensors              Pass: 59                   Fail: 0                   
TestSuite: syslog               Pass: 79                   Fail: 0                   

----------------------------------------------------------------------------------------------------
Total Pass: 2300                 Total Fail: 0                   

**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
